packages/block-editor/src/components/block-bindings-indicator/index.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * WordPress dependencies
+ */
+import { ToolbarItem, ToolbarGroup, Icon } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { connection } from '@wordpress/icons';
+import { _x } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../store';
+
+export default function BlockBindingsIndicator( { clientId } ) {
+	const isConnected = useSelect(
```

Comment:
There is now more convenient and performant way to access block bindings for the block:

https://github.com/WordPress/gutenberg/blob/7eb0a5acb3f8ab7e8edb30e26934e25477fb1b67/packages/block-editor/src/components/rich-text/index.js#L121-L123

Reply:
This block bindings aren't readily present in the block edit context for the Block Bindings Indicator component, unfortunately.

I believe this is because the context is is created in [Block Edit](https://github.com/WordPress/gutenberg/blob/fix/update-block-bindings-ui-indicator/packages/block-editor/src/components/block-edit/index.js), which is instantiated in [Block List](https://github.com/WordPress/gutenberg/blob/fix/update-block-bindings-ui-indicator/packages/block-editor/src/components/block-list/index.js), whereas the Block Bindings Indicator is found outside of that scope in the block toolbar — though I may be misunderstanding here, or perhaps it's possible to pass the context to the indicator somehow.

Should I look into trying to make the context available here and/or keep digging into this?

Reply:
Yes, that makes sense since this new toolbar item gets added directly to the block toolbar. All other toolbar items are usually injected as fills so they have access to the block edit context. I missed that important nuance, so it's perfectly fine as is.

Reply:
I believe we could/should move the logic to check if the block is connected to [the toolbar component](https://github.com/WordPress/gutenberg/pull/59185/files#diff-9ac2a4466e7c0a85aaff0b40c5d7bc64c894132528ca3c879db4bebb438617dcR170). It is the way the rest of components seems to be used.

I mean exporting this component as it is, with the Toolbar Item, and load it conditionally adding a `isUsingBindings` or something similar to [the Toolbar useSelect logic](https://github.com/WordPress/gutenberg/blob/9d6f97c42fcc5447f0e1b38bc413ab3db9513cc5/packages/block-editor/src/components/block-toolbar/index.js#L60-L113). That way, we could just load it conditionally by doing:

```js
{ isUsingBindings && <BlockBindingsIndicator /> }
```

What do you think?